### PR TITLE
[Feature] Dispatch DiscreteSAC loss module

### PR DESCRIPTION
**Pull request by Blonck**
## Description

Enable dispatching arguments for the `.forward()` method of the DiscreteSAC loss module.

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: torchrl/objectives/sac.py [hunk 1]
```diff
@@ -713,6 +713,111 @@ class DiscreteSACLoss(LossModule):
...
+    >>> batch = [2, ]
+    >>> action = spec.rand(batch)
+    >>> _ = loss.select_out_keys("loss_actor", "loss_qvalue")
+    >>> loss_actor, loss_qvalue = loss(
+    ...     observation=torch.randn(*batch, n_obs),
+    ...     action=action,
+    ...     next_done=torch.zeros(*batch, 1, dtype=torch.bool),
+    ...     next_observation=torch.zeros(*batch, n_obs),
+    ...     next_reward=torch.randn(*batch, 1))
+    >>> loss_actor.backward()
```
Comment (vmoens) [inline torchrl/objectives/sac.py:820]:
```suggestion
        >>> import torch
        >>> from torch import nn
        >>> from torchrl.data.tensor_specs import OneHotDiscreteTensorSpec
        >>> from torchrl.modules.distributions.continuous import NormalParamWrapper
        >>> from torchrl.modules.distributions.discrete import OneHotCategorical
        >>> from torchrl.modules.tensordict_module.actors import ProbabilisticActor, ValueOperator
        >>> from torchrl.modules.tensordict_module.common import SafeModule
        >>> from torchrl.objectives.sac import DiscreteSACLoss
        >>> n_act, n_obs = 4, 3
        >>> spec = OneHotDiscreteTensorSpec(n_act)
        >>> net = NormalParamWrapper(nn.Linear(n_obs, 2 * n_act))
        >>> module = SafeModule(net, in_keys=["observation"], out_keys=["logits"])
        >>> actor = ProbabilisticActor(
        ...     module=module,
        ...     in_keys=["logits"],
        ...     out_keys=["action"],
        ...     spec=spec,
        ...     distribution_class=OneHotCategorical)
        >>> class ValueClass(nn.Module):
        ...     def __init__(self):
        ...         super().__init__()
        ...         self.linear = nn.Linear(n_obs, n_act)
        ...     def forward(self, obs):
        ...         return self.linear(obs)
        >>> module = ValueClass()
        >>> qvalue = ValueOperator(
        ...     module=module,
        ...     in_keys=['observation'])
        >>> loss = DiscreteSACLoss(actor, qvalue, num_actions=actor.spec["action"].space.n)
        >>> batch = [2, ]
        >>> action = spec.rand(batch)
        >>> _ = loss.select_out_keys("loss_actor", "loss_qvalue")
        >>> loss_actor, loss_qvalue = loss(
        ...     observation=torch.randn(*batch, n_obs),
        ...     action=action,
        ...     next_done=torch.zeros(*batch, 1, dtype=torch.bool),
        ...     next_observation=torch.zeros(*batch, n_obs),
        ...     next_reward=torch.randn(*batch, 1))
        >>> loss_actor.backward()
```

Comment (vmoens) [inline torchrl/objectives/sac.py:820]: I would add a comment before that

File: torchrl/objectives/sac.py [hunk 2]
```diff
@@ -831,8 +944,44 @@ def _forward_value_estimator_keys(self, **kwargs) -> None:
...
+    def in_keys(self):
+        keys = [
+            self.tensor_keys.action,
+            ("next", self.tensor_keys.reward),
+            ("next", self.tensor_keys.done),
+            *self.actor_network.in_keys,
+            *[("next", key) for key in self.actor_network.in_keys],
+            *self.qvalue_network.in_keys,
+        ]
+        return list(set(keys))
```
Comment (vmoens) [inline torchrl/objectives/sac.py:961]: just realizing that
this is pretty slow, and in_keys will be called often. I would memoize that with an `_in_keys` like we do for `out_keys`

**Review by vmoens — COMMENTED** [review]
LGTM 
Some suggestions

File: torchrl/objectives/sac.py (hunk 2 above)
Comment (Blonck) [reply torchrl/objectives/sac.py:961]: Good point, I will port this back to the other `in_keys` of the loss modules. 
It's worth noting, however, that each time the .set_keys() method is invoked, we will also need to update the _in_keys accordingly.

Comment (vmoens) [reply torchrl/objectives/sac.py]: we could have `set_keys()` destroy `_in_keys`

**Review by vmoens — APPROVED** [review]
